src/main/java/org/commcare/formplayer/repo/impl/PostgresFormSessionRepo.java

```diff
@@ -328,4 +305,33 @@ public String replaceTableName(String query){
         return String.format(query, Constants.POSTGRES_SESSION_TABLE_NAME);
     }
 
+    public int purgeFormSessions() {
+        // Modeled on https://stackoverflow.com/a/6730401/2820312
+        int deletedRows = 0;
+        try {
+            String createFuncQuery = "create or replace function custom_safe_cast(text,anyelement) \n" +
+                    "returns anyelement \n" +
+                    "language plpgsql as $$ \n" +
+                    "begin \n" +
+                    "    $0 := $1; \n" +
+                    "    return $0; \n" +
+                    "    exception when others then \n" +
+                    "        return $2; \n" +
+                    "end; $$;";
+            this.jdbcTemplate.execute(createFuncQuery);
+            String deleteQuery = replaceTableName(
+                    "delete from %s where custom_safe_cast(dateopened, '2011-01-01'::timestamp) < NOW() - INTERVAL '7 days';"
+            );
+            log.info("Beginning state form session purge");
+            long start = System.currentTimeMillis();
+            deletedRows = this.jdbcTemplate.update(deleteQuery);
+            long elapsed = System.currentTimeMillis() - start;
```

Comment:
Suggestion: This timing might be valuable to report back to DataDog with the event information.